parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -403,6 +406,16 @@ pub mod pallet {
 				body_root: body_root_hash,
 			};
 
+			let beacon_block_root: H256 = merkleization::hash_tree_root_beacon_header(header.clone())
+				.map_err(|_| DispatchError::Other("Beacon header hash tree root failed"))?.into();
+			if beacon_block_root != update.block_root {
```

Comment:
We send the block hash from the relayer, while the block root calculated here is from the merkleization of the beacon block, which means the calculated one could be correct and the hash sent from the relayer could be incorrect. I used this mainly to debug problems with the beacon block body merkleization, but that works now.

Would it be OK if I then rather log a ticket to remove the hashes sent from the relayer?

Reply:
For the block root calculated in the relayer: do we use that hash in other parts of the relayer? If so, it might be worth adding tests to ensure that the block root calculation aligns across the relayer and parachain.

We can't trust hashes sent from the relayer, so if we're calculating them anyway I think we're better off not sending them. Fewer things to keep track of.